aries_cloudagent/protocols/didexchange/v1_0/manager.py

```diff
@@ -606,6 +629,17 @@ async def create_response(
             async with self.profile.session() as session:
                 wallet = session.inject(BaseWallet)
                 my_info = await wallet.get_local_did(conn_rec.my_did)
+            did = my_info.did
+        elif use_public_did:
+            async with self.profile.session() as session:
+                wallet = session.inject(BaseWallet)
+                my_info = await wallet.get_public_did()
+            if not my_info:
+                raise DIDXManagerError("No public DID configured")
+            conn_rec.my_did = my_info.did
+            did = my_info.did
+            if not did.startswith("did:"):
+                did = f"did:sov:{did}"
```

Comment:
do public unqualified did's exist? is this how we should handle them?

Reply:
All Indy style DIDs in ACA-Py's wallet are stored unqualified. This includes public Indy DIDs so this check and prepending `did:sov:` is necessary for the DID that we send in the request/response is fully qualified.

When we directly support did:indy DIDs, they should be properly stored with their method identifier.